api/test/unit/validation/attributeValidator.test.js

```diff
@@ -0,0 +1,101 @@
+      });
+    });
+    it('Should fail to validate the Policy with overlapping time range in specific date',function(){
+      fakePolicy.schedules.specific_date[1].start_date_time = '2015-06-04T10:53';
+      fakePolicy.schedules.specific_date[1].end_date_time = '2015-06-21T23:45';
+      attributeValidator.validatePolicy(fakePolicy,function(result){
+        expect(result[0].property).to.equal('specific_date.start_date_time');
+        expect(result[0].stack).to.equal('Date range of specific_date[1] is overlapped with date range of specific_date[0]');
+        expect(result[0].message).to.equals('specific_date.start_date_time and specific_date.end_date_time ranges are overlapping');
+      });
+    });
+    it('should fail to validate policy with overlapping time range in days of week in recurring schedule',function(){
+      fakePolicy.schedules.recurring_schedule[2].days_of_week = [4,5,2];
+      attributeValidator.validatePolicy(fakePolicy,function(result){
+        expect(result[0].property).to.equal('recurring_schedule.start_time');
+        expect(result[0].message).to.equal('recurring_schedule.start_time and recurring_schedule.end_time ranges are overlapping');
+        expect(result[0].stack).to.equal('Days_of_week based time range of recurring_schedule[0] is overlapped with time range of recurring_schedule[2]');
+      });
+    });
+    it('should fail to validate policy with overlapping time range in days of month in recurring schedule',function(){
+      fakePolicy.schedules.recurring_schedule[1].days_of_month = [12,20,24];
+      attributeValidator.validatePolicy(fakePolicy,function(result){
+        expect(result[0].property).to.equal('recurring_schedule.start_time');
+        expect(result[0].message).to.equal('recurring_schedule.start_time and recurring_schedule.end_time ranges are overlapping');
+        expect(result[0].stack).to.equal('Days_of_month based time range of recurring_schedule[1] is overlapped with time range of recurring_schedule[3]');
+      });
+    });
+    it('should validate the policy successfully if days of month in recurring schedule is overlapping but start_time and end_time in overlapped date are non-overlapping',function(){
+      fakePolicy.schedules.recurring_schedule[1].days_of_week = [4,5,2];
+      fakePolicy.schedules.recurring_schedule[1].start_time = '19:00'
+      fakePolicy.schedules.recurring_schedule[1].end_time = '23:15';
+      attributeValidator.validatePolicy(fakePolicy,function(result){
+        expect(result).to.be.empty;
+      });
+    });
+    it('should fail to validate the policy if start_time is after end_time in recurring schedule',function(){
+      fakePolicy.schedules.recurring_schedule[1].start_time = '23:00'
+      fakePolicy.schedules.recurring_schedule[1].end_time = '13:15';
+      attributeValidator.validatePolicy(fakePolicy,function(result){
+        expect(result).to.not.be.null;
```

Comment:
Why are we checking for NOT NULL in this test and not in the other tests? Anyways, do we need this check?